domain/operation/state/start.go

```diff
@@ -375,11 +348,18 @@ VALUES ($insertOperation.*)
 	return errors.Capture(tx.Query(ctx, stmt, args).Run())
 }
 
-func (s *State) insertOperationParameter(ctx context.Context, tx *sqlair.TX, operationUUID, key, value string) error {
+func (s *State) insertOperationParameter(ctx context.Context, tx *sqlair.TX, operationUUID, key string,
+	value any) error {
+	insertValue := fmt.Sprintf("%v", value)
+	// quote strings to keep them as strings in the database and not try to
+	// decode them on fetch
+	if v, ok := value.(string); ok {
+		insertValue = fmt.Sprintf("%q", v)
+	}
```

Comment:
I think it would be better to have proper encode/decode functions here, to handle the various cases we support
e.g.
https://github.com/juju/juju/blob/524877433fac3642f31456edda78999396fba10a/domain/application/config.go#L156-L186

Reply:
Here, and elsewhere